radicle-node/src/runtime/handle.rs

```diff
@@ -58,12 +58,65 @@ impl<T> From<chan::SendError<T>> for Error {
     }
 }
 
+/// Event enum to emit.
+#[derive(Clone)]
+pub enum Event {
```

Comment:
Two notes:

- The [Event](https://github.com/radicle-dev/heartwood/blob/master/radicle-node/src/service.rs#L83) you showed here seems to be the type `node` sends to reactor which is an interface to network. We might not necessarily want the same type for local events.

- This can be a struct, but the downside is a lot of impl would change, etc. So I went with an `enum` here and implement `From` in order to get eq. of `.type`.

Reply:
It sends it to reactor because the thought was that the reactor would then send it to the user via an `Io::Event`. There is an advantage in consolidating all of the effects of service under a single enum (`reactor::Io`).

Reply:
Oh, also, we can use `Event` for subscribing as well, but I thought it wouldn't pass the review. So I created two types, `Event` and `EventType`, but if we can look past that, we can merge them into one, and use empty `Event` as event_type for subscribe requests as well.

Reply:
Not sure I follow -- there shouldn't need to be a subscribe request event.